--- src/yafu.pl	2018-06-23 19:03:09.628834400 +0000
+++ bin/yafu.pl	2018-06-23 19:04:08.969016800 +0000
@@ -3,9 +3,18 @@
 use warnings;
 use strict;
 use LWP::Simple;
+use LWP::UserAgent;
 
 my %total;
 
+my $user="YOURuser";
+my $password="YOURpw";
+
+# login to factordb
+my $ua       = LWP::UserAgent->new();
+$ua->cookie_jar( {} );
+my $response = $ua->post("http://factordb.com/login.php", { 'user' => $user, 'pass' => $password, 'dlogin' => "Login" } );
+
 while(1){
     print "get composites\n";
     my $rand=int(rand(100));
@@ -34,9 +43,12 @@
 	    if ( scalar(@results) > 0 ) {
 		print "===========================================================================\n";
 		print "report factors\n";
+		# login again
+		$response = $ua->post("http://factordb.com/login.php", { 'user' => $user, 'pass' => $password, 'dlogin' => "Login" } );
 		my $url="http://factordb.com/report.php?report=".$composite."%3D".join('*',@results);
 		#print "$url\n";
-		$contents=get($url);
+		$response = $ua->get($url);
+		$contents = $response->decoded_content();
 		#print "$contents\n";
 
 		my $nofactors     = ($contents =~ s/Does not divide//g);
