--- aliqueit.cc.orig	2018-06-23 15:01:14.000000000 +0000
+++ aliqueit.cc	2018-06-23 15:42:14.358719500 +0000
@@ -901,7 +901,7 @@
 	fo << post_data;
 	fo.close();
 	cout << "Sending " << sent_lines << " lines..." << endl;
-	system( ( "wget --cache=off --output-document=" + tmp_file + " --post-file=" + post_file + " http://factorization.ath.cx/search.php?report=true" ).c_str() );
+	system( ( "wget --cache=off --output-document=" + tmp_file + " --post-file=" + post_file + " http://factordb.com/report.php" ).c_str() );
 	delete_file( post_file.c_str() );
 
 	f.clear();
